homeassistant/components/playstation_network/config_flow.py

```diff
@@ -0,0 +1,116 @@
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    try:
+        npsso: str = str(data.get("pin"))
+        psn = PSNAWP(npsso)
+    except PSNAWPAuthenticationError as error:
+        raise ConfigEntryAuthFailed(error) from error
+    except Exception as ex:
+        raise ConfigEntryNotReady(ex) from ex
+
+    try:
+        user = await hass.async_add_executor_job(lambda: psn.user(online_id="me"))
+        client = psn.me()
+        coordinator = PlaystationNetworkCoordinator(hass, psn, user, client)
+        await coordinator.async_config_entry_first_refresh()
+    except PSNAWPAuthenticationError as error:
+        raise ConfigEntryAuthFailed(error) from error
+    except Exception as ex:
+        raise ConfigEntryNotReady(ex) from ex
+
+    # Return info that you want to store in the config entry.
+    return {
+        "title": "Playstation Network",
+        "npsso": npsso,
+        "username": coordinator.data.get("username"),
+        "data": data,
+    }
+
+
+class PlaystationNetworkConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Playstation Network."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self) -> None:
+        """Playstation Network Config Flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is None or user_input == {}:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            )
+
+        try:
+            info = await validate_input(self.hass, user_input)
+            await self._async_set_unique_id_and_abort_if_already_configured(
+                info[CONF_USERNAME]
+            )
+        except CannotConnect:
+            errors = {"base": "Cannot Connect"}
+        except InvalidAuth:
+            errors = {"base": "Invalid Authentication"}
+        except AbortFlow:
+            errors = {"base": "This account is already configured"}
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title=info["title"], data=info)
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def _async_set_unique_id_and_abort_if_already_configured(
+        self, unique_id: str
+    ) -> None:
+        """Set the unique ID and abort if already configured."""
+        await self.async_set_unique_id(unique_id, raise_on_progress=False)
+        self._abort_if_unique_id_configured(
+            updates={CONF_USERNAME: unique_id},
+        )
```

Comment:
```suggestion
        await self.async_set_unique_id(unique_id)
        self._abort_if_unique_id_configured()
```
I think username is not acceptable as unique id, as it can be changed. Use instead the account_id instead of online id, as that is unique and immutable in the PSN